File: mobile/ios/Accord/Models/Node.swift
// Node.swift — A community/server (called "Node" in Accord)

import Foundation

struct Node: Identifiable, Codable, Hashable {
    let id: String
    var name: String
    var iconURL: URL?
    var description: String?
    var ownerId: String
    var channels: [Channel] = []
    var members: [User] = []
}

File: mobile/android/app/src/main/java/com/accord/data/model/Node.kt
package com.accord.data.model

data class Node(
    val id: String,
    val name: String,
    val description: String = "",
    val iconUrl: String? = null,
    val ownerId: String,
    val memberCount: Int = 0,
    val categories: List<Category> = emptyList(),
)

data class Category(
    val id: String,
    val name: String,
    val position: Int = 0,
)
